internal/clients/vault_secrets_preview.go

```diff
@@ -157,11 +160,12 @@ func CreateMongoDBAtlasRotatingSecret(
 }
 
 // CreateAwsIntegration NOTE: currently just needed for tests
-func CreateAwsIntegration(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, name, roleArn string) (*secretmodels.Secrets20231128AwsIntegration, error) {
+func CreateAwsIntegration(ctx context.Context, client *Client, loc *sharedmodels.HashicorpCloudLocationLocation, name, roleArn, audience string, capabilities []*secretmodels.Secrets20231128Capability) (*secretmodels.Secrets20231128AwsIntegration, error) {
 	body := secretmodels.SecretServiceCreateAwsIntegrationBody{
-		Name: name,
+		Name:         name,
+		Capabilities: capabilities,
 		FederatedWorkloadIdentity: &secretmodels.Secrets20231128AwsFederatedWorkloadIdentityRequest{
-			Audience: loc.OrganizationID,
+			Audience: audience,
```

Comment:
Hardcoding the orgID as the audience was a wrong assumption I introduced when implementing the dynamic secret data source.